File: go/http/cmd/healthcheck.go
// SPDX-FileCopyrightText: © 2024 Sebastian Davids <sdavids@gmx.de>
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"cmp"
	"io"
	"log/slog"
	"net/http"
	"os"
	"runtime"
	"time"
)

func main() {
	url := cmp.Or(os.Getenv("HEALTHCHECK_URL"), "http://localhost:3000/-/health/liveness")
	client := &http.Client{
		Timeout: time.Second * 5,
	}
	res, err := client.Get(url) //nolint
	if err != nil {
		slog.Error("", slog.Any("error", err))
		os.Exit(69) // EX_UNAVAILABLE
	}
	defer func(Body io.ReadCloser) {
		_ = Body.Close()
	}(res.Body)
	if res.StatusCode == http.StatusOK {
		defer os.Exit(0) // EX_OK
	} else {
		defer os.Exit(100)
	}
	runtime.Goexit()
}

File: java/http/src/HealthCheck.java
/*
 * SPDX-FileCopyrightText: © 2024 Sebastian Davids <sdavids@gmx.de>
 * SPDX-License-Identifier: Apache-2.0
 */
import static java.util.Objects.requireNonNullElse;

void main() {
  var u =
    requireNonNullElse(System.getenv("HEALTHCHECK_URL"), "http://localhost:3000/-/health/liveness");

  URL url = null;
  try {
    url = URI.create(u).toURL();
  } catch (IllegalArgumentException | MalformedURLException e) {
    System.err.printf("'%s' is not a valid URL%n", u);
    System.exit(64); // EX_USAGE
  }

  try {
    var connection = (HttpURLConnection) url.openConnection();
    connection.setConnectTimeout(5000);
    connection.setReadTimeout(5000);

    connection.disconnect();

    System.exit(connection.getResponseCode() == HttpURLConnection.HTTP_OK ? 0 : 100);
  } catch (IOException e) {
    System.err.printf("%s%n", u);
    e.printStackTrace();
    System.exit(70); // EX_SOFTWARE
  }
}
